### ISSUE TITLE
TCH001 TCH002 false negative – add msgspec.Struct support

### ISSUE THREAD

**Issue by Olegt0rr** [astral-sh/ruff#6571]
> TCH001 [*] Move application import `.objects.Issue` into a type-checking block

and 

> TCH002 [*] Move third-party import `myapp.types.Issue` into a type-checking block

```python
from __future__ import annotations

from msgspec import Struct

from myapp.types import Issue


class FullIssue(Struct):
    parent: Issue | None = None

```

Ruff v0.0.284 
target-version = "py39"

**Comment by charliermarsh** [comment]
Does it work as expected if you set [`runtime-evaluated-base-classes`](https://beta.ruff.rs/docs/settings/#flake8-type-checking-runtime-evaluated-base-classes)?

```toml
[tool.ruff.flake8-type-checking]
runtime-evaluated-base-classes = ["msgspec.Struct"]
```

**Comment by Olegt0rr** [comment]
Yes, thanks!

But as long it's a default behavior for all msgspec structs – it would be nice to make similar support as pydantic

**Comment by charliermarsh** [comment]
You can add `pydantic.BaseModel` to that list too. I'm considering adding them by default.

**Comment by Olegt0rr** [comment]
Aw, the same setting has troubles with inherits

```python
from __future__ import annotations

from msgspec import Struct

from yatracker.types import Issue


class Foo(Struct):
    """Base"""


class FullIssue(Foo):
    parent: Issue | None = None

```

**Comment by charliermarsh** [comment]
Yes, this is a known issue. It'll be solved at some point, but probably not in an upcoming release, because it requires us to resolve symbols like `Foo` across files to really be useful in practice (e.g., if you did `from module import Foo`, and `Foo` inherited from `Struct`, we'd need to detect that).

You can either add every class to the list, or include `Struct` as a base class, like:

```python
from __future__ import annotations

from msgspec import Struct

from yatracker.types import Issue


class Foo(Struct):
    """Intermediate class"""


class FullIssue(Foo, Struct):
    parent: Issue | None = None
```

Or just disable these rules when working with libraries that rely on type annotations at runtime.
